src/event.rs

```diff
@@ -74,9 +74,94 @@ pub enum Event<'a, T: 'static> {
     UserEvent(T),
 
     /// Emitted when the application has been suspended.
+    ///
+    /// # Portability
+    ///
+    /// Not all platforms support the notion of suspending applications, and there may be no
+    /// technical way to guarantee being able to emit a `Suspended` event if the OS has
+    /// no formal application lifecycle (currently only Android and iOS do). For this reason
+    /// Winit does not currently try to emit pseudo `Suspended` events before the application
+    /// quits on platforms without an application lifecycle.
+    ///
+    /// In case of corner cases with how platform-specific events are mapped to Winit `Suspended`
+    /// and `Resumed` events it's recommended that applications be able to gracefully handle redundant
+    /// (i.e. back to back) `Suspended` or `Resumed` events.
```

Comment:
Is this currently something we're bothered by?

Reply:
It came up in the discussion of https://github.com/rust-windowing/winit/issues/2185 - so it hopefully helps clarify what applications can (or can not rely on).

Some specific, plausible corner cases were also discussed there, whereby I feel like it would be wise for Winit to not over-promise that there will never be unforseen corner cases (considering that it seems like it shouldn't be a significant burden for applications to be prepared to handle redundant events).

The alternative imho would be to explicitly promise that Winit will never emit redundant events (otherwise the documentation isn't helping clarify what applications can rely on).

In practice I'd expect Winit to strive to avoid any redundant events, but by under promising I think there's a better chance that apps will cope fine if they do hit some kind of corner case.